delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/client/ElasticSearchClient.scala

```diff
@@ -213,16 +223,25 @@ class ElasticSearchClient(client: HttpClient, endpoint: Uri)(implicit as: ActorS
   )(
       sort: SortList = SortList.empty
   ): HttpResult[Json] = {
-    val searchEndpoint = (endpoint / indexPath(indices) / searchPath).withQuery(Uri.Query(defaultQuery ++ qp.toMap))
-    val payload        = QueryBuilder(query).withSort(sort).withTotalHits(true).build
-    client.toJson(Post(searchEndpoint, payload))
+    if (indices.isEmpty)
+      IO.pure(emptyResults)
+    else {
+      val (indexPath, q) = indexPathAndQuery(indices, QueryBuilder(query))
+      val searchEndpoint = (endpoint / indexPath / searchPath).withQuery(Uri.Query(defaultQuery ++ qp.toMap))
+      val payload        = q.withSort(sort).withTotalHits(true).build
+      client.toJson(Post(searchEndpoint, payload))
+    }
   }
 
   private def discardEntity(resp: HttpResponse) =
     UIO.delay(resp.discardEntityBytes()) >> IO.unit
 
-  private def indexPath(indices: Set[String]): String =
-    if (indices.isEmpty) allIndexPath else indices.mkString(",")
+  private def indexPathAndQuery(indices: Set[String], query: QueryBuilder): (String, QueryBuilder) = {
+    indices.toList match {
```

Comment:
I don't think we should always use the query to filter out the indices. We should only do it if the number of indices is higher than a threshold. It is way faster to only query a subset of indices than to query all and filter afterwards

Reply:
What should we use as the threshold? The whole fist line of the HTTP request has to be under 4kb [http.max_initial_line_length](https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-network.html).
It's not only path but also HTTP protocol parts`GET / HTTP/1.0` and query params. 
Should we make it configurable (default to 3072), and if `indices.mkString(",")` exceeds that number, than use the filtering?

Reply:
I'd say a threshold of 15-20 indices should be on the safe side for us. 

Notice that the aggregate views are limited by configuration (and the default value is around 10 If I remember well). That means that when using aggregated views we actually will always filter the indices on the URL instead of the payload, which will be better.